cpp/src/arrow/array/builder_base.h

```diff
@@ -250,6 +250,9 @@ class ARROW_EXPORT ArrayBuilder {
   // Set the next validity bits to null (i.e. invalid).
   void UnsafeSetNull(int64_t length);
 
+  /// Set the index bit to null or not null
+  void UnsafeSetIsNull(int64_t index, bool is_null);
```

Comment:
Looks this api doesn't work for variable length types like string. Probably should not be in base class.

Reply:
Thanks. Yeah it makes sense. I moved this method to `NumericBuilder`